src/v/storage/offset_translator_data.h

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "model/fundamental.h"
+#include "serde/serde.h"
+
+#include <absl/container/btree_map.h>
+
+namespace storage {
+
+/// Provides offset translation between raw log offsets and offsets not counting
+/// filtered batches (basically our internal batch types that we write into the
+/// data partitions). This is needed because even though our internal batch
+/// types are filtered out when sent to clients, kafka clients are not prepared
+/// for these batches to occupy offset space (see
+/// https://github.com/vectorizedio/redpanda/issues/1184 for details).
+///
+/// It works by maintaining an in-memory map of all filtered batch offsets.
+/// This map allows us to quickly find a delta between the raw log offset and
+/// corresponding translated offset.
+class offset_translator_data {
```

Comment:
small nit: usually `state` suffix is used across the code base to denote set of data maintained in the component